modules/porous_flow/src/userobjects/AdvectiveFluxCalculatorBase.C

```diff
@@ -847,3 +841,273 @@ AdvectiveFluxCalculatorBase::PQPlusMinus(dof_id_type sequential_i,
 
   return result;
 }
+
+void
+AdvectiveFluxCalculatorBase::buildCommLists()
+{
+  /**
+   * Build the multi-processor communication lists.
+   *
+   * (A) We will have to send _u_nodal information to other processors.
+   * This is because although we can Evaluate Variables at all elements in
+   * _fe_problem.getEvaluableElementRange(), in the PorousFlow setting
+   * _u_nodal could depend on Material Properties within the elements, and
+   * we can't access those Properties within the ghosted elements.
+   * We don't know which processors require _u_nodal from our nodes.
+   * However, we do know which nodes we require information about, and the
+   * processors that can give us that information.  So we store that in
+   * _nodes_to_receive.  Then we advertise to those processors that we need
+   * the information.  Similarly, we will receive advertisements from the
+   * other processors saying they need information from us.  We then store
+   * that in _nodes_to_send.
+   *
+   * (B) We need to send _kij information to other processors.
+   * The same strategy is followed, first building pairs_to_receive and then
+   * _pairs_to_send.
+   */
+
+  // We use this (hopefully unique) message tag for all communication in this method.
+  // QUESTION: If there are multiple AdvectiveFluxCalculatorBase in the input file, will this
+  // work?
+  Parallel::MessageTag send_tag = _communicator.get_unique_tag(4712);
```

Comment:
Did you just increment my choice of tag here? It was based on
![image](https://user-images.githubusercontent.com/202302/57098752-0c809580-6cd8-11e9-8f3d-3f2ce0d7829a.png)

Reply:
Yea, i just incremented your choice.  I was trying to think "4" was "D" (for Daniel) but then couldn't get the "7", "11, or "71", etc match anything sensible.